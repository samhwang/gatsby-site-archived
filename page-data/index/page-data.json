{"componentChunkName":"component---src-pages-index-tsx","path":"/","result":{"pageContext":{}},"staticQueryHashes":["2956700167","4024886621","971228977"]}