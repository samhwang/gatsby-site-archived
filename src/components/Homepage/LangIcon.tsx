import React from 'react';

function LangIcon(props: { name: string }) {
  const { name } = props;

  return (
    <li className="list-inline-item">
      <i className={`fab fa-${name}`} />
    </li>
  );
}

export default LangIcon;
