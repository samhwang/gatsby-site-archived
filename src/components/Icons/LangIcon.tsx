import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
// eslint-disable-next-line no-unused-vars
import { IconPrefix, IconName } from '@fortawesome/fontawesome-svg-core';

type LangIconProps = {
  name: string;
};

function LangIcon({ name }: LangIconProps) {
  const iconProps = ['fab', name] as [IconPrefix, IconName];

  return (
    <li className="list-inline-item">
      <FontAwesomeIcon icon={iconProps} />
    </li>
  );
}

export default LangIcon;
