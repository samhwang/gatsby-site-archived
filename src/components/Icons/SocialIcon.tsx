import type { FC } from 'react';
import { OutboundLink } from 'gatsby-plugin-google-gtag';
import { createStyles, makeStyles } from '@material-ui/core';
import {
  LinkedIn as LinkedInIcon,
  GitHub as GithubIcon,
  Email as EmailIcon,
  CloudDownload as CloudDownloadIcon,
} from '@material-ui/icons';
import type { SvgIconTypeMap, Theme } from '@material-ui/core';
import type { OverridableComponent } from '@material-ui/core/OverridableComponent';

export interface SocialLinkProps {
  name: string;
  url: string;
}

type MaterialIconType = OverridableComponent<SvgIconTypeMap<{}, 'svg'>>;

function getIconFromName(name: string): MaterialIconType | undefined {
  // eslint-disable-next-line default-case
  switch (name.toLowerCase()) {
    case 'github':
      return GithubIcon;

    case 'linkedin':
      return LinkedInIcon;

    case 'pdf':
      return CloudDownloadIcon;

    case 'email':
      return EmailIcon;

    default:
      return undefined;
  }
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    icon: {
      fontSize: theme.spacing(10),
      margin: theme.spacing(3),
      color: '#cccccc',
      '&:hover': {
        color: theme.palette.primary.main,
      },
    },
  })
);

const SocialIcon: FC<SocialLinkProps> = ({ name, url }) => {
  const TheIcon = getIconFromName(name);
  const classes = useStyles();

  if (!TheIcon) {
    return <></>;
  }

  return (
    <OutboundLink href={url} alt={`${name} link`}>
      <TheIcon className={classes.icon} />
    </OutboundLink>
  );
};

export default SocialIcon;
