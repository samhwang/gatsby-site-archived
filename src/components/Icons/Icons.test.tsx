/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import renderer from 'react-test-renderer';
import './index';
import LangIcon from './LangIcon';
import SocialIcon from './SocialIcon';

describe('Lang Icon', () => {
  it('Should render without crashing and match snapshot', () => {
    const tree = renderer.create(<LangIcon name="php" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Social Icon', () => {
  it('Should render without crashing and match snapshot', () => {
    const tree = renderer
      .create(
        <SocialIcon
          icon="fab github"
          name="Github"
          url="https://github.com/samhwang"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
