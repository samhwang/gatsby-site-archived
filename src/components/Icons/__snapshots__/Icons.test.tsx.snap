// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lang Icon Should render without crashing and match snapshot 1`] = `
<li
  className="list-inline-item"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-php fa-w-20 "
    data-icon="php"
    data-prefix="fab"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 640 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M320 104.5c171.4 0 303.2 72.2 303.2 151.5S491.3 407.5 320 407.5c-171.4 0-303.2-72.2-303.2-151.5S148.7 104.5 320 104.5m0-16.8C143.3 87.7 0 163 0 256s143.3 168.3 320 168.3S640 349 640 256 496.7 87.7 320 87.7zM218.2 242.5c-7.9 40.5-35.8 36.3-70.1 36.3l13.7-70.6c38 0 63.8-4.1 56.4 34.3zM97.4 350.3h36.7l8.7-44.8c41.1 0 66.6 3 90.2-19.1 26.1-24 32.9-66.7 14.3-88.1-9.7-11.2-25.3-16.7-46.5-16.7h-70.7L97.4 350.3zm185.7-213.6h36.5l-8.7 44.8c31.5 0 60.7-2.3 74.8 10.7 14.8 13.6 7.7 31-8.3 113.1h-37c15.4-79.4 18.3-86 12.7-92-5.4-5.8-17.7-4.6-47.4-4.6l-18.8 96.6h-36.5l32.7-168.6zM505 242.5c-8 41.1-36.7 36.3-70.1 36.3l13.7-70.6c38.2 0 63.8-4.1 56.4 34.3zM384.2 350.3H421l8.7-44.8c43.2 0 67.1 2.5 90.2-19.1 26.1-24 32.9-66.7 14.3-88.1-9.7-11.2-25.3-16.7-46.5-16.7H417l-32.8 168.7z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</li>
`;

exports[`Social Icon Should render without crashing and match snapshot 1`] = `
<a
  alt="Github link"
  href="https://github.com/samhwang"
  onClick={[Function]}
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-github fa-w-16 "
    data-icon="github"
    data-prefix="fab"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 496 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</a>
`;
